grype/db/v5/store/store.go

```diff
@@ -99,7 +99,13 @@ func (s *store) GetVulnerabilityNamespaces() ([]string, error) {
 func (s *store) GetVulnerability(namespace, id string) ([]v5.Vulnerability, error) {
 	var models []model.VulnerabilityModel
 
-	result := s.db.Where("namespace = ? AND id = ?", namespace, id).Find(&models)
+	var query = s.db
+
+	if namespace != "" {
+		query = query.Where("namespace = ?", namespace)
+	}
+
+	result := query.Where("id = ?", id).Find(&models)
```

Comment:
nit: this query builder might be slightly more clear if the ID clause is added on line 102, e.g.:
```
	query := s.db.Where("id = ?", id)

	if namespace != "" {
		query = query.Where("namespace = ?", namespace)
	}

	result := query.Find(&models)
```

Reply:
fixed

Reply:
fixed